scripts/check-gmp.gmp

```diff
@@ -0,0 +1,1565 @@
+        if script_args.gmp_username:
+            print("GMP_User: %s" % script_args.gmp_username)
+        if script_args.task:
+            print_without_pipe("Task: %s" % script_args.task)
+
+    end_session(
+        im,
+        "|High=%i Medium=%i Low=%i" % (high_count, medium_count, low_count),
+        ret,
+    )
+
+
+def retrieve_nvt_data(result):
+    """Retrieve the nvt data out of the result object
+
+    This function parse the xml tree to find the important nvt data.
+
+    Arguments:
+        result {lxml object} -- Result xml object
+
+    Returns:
+        Tuple -- List with oid, name, desc, port and dfn
+    """
+    oid = result.xpath("nvt/@oid")
+    name = result.xpath("nvt/name/text()")
+    desc = result.xpath("description/text()")
+    port = result.xpath("port/text()")
+
+    if oid:
+        oid = oid[0]
+
+    if name:
+        name = name[0]
+
+    if desc:
+        desc = desc[0]
+    else:
+        desc = ""
+
+    if port:
+        port = port[0]
+    else:
+        port = ""
+
+    certs = result.xpath("nvt/cert/cert_ref")
+
+    dfn_list = []
+    for ref in certs:
+        ref_type = ref.xpath("@type")[0]
+        ref_id = ref.xpath("@id")[0]
+
+        if ref_type in "DFN-CERT":
+            dfn_list.append(ref_id)
+
+    return (oid, name, desc, port, dfn_list)
+
+
+def print_nvt_data(nvts):
+    """Print nvt data
+
+    Prints for each nvt found in the array the relevant data
+
+    Arguments:
+        nvts {object} -- Object holding all nvts
+    """
+    for key, nvt_data in nvts.items():
+        if key is "log" and not script_args.showlog:
+            continue
+        for nvt in nvt_data:
+            print_without_pipe("NVT: %s (%s) %s" % (nvt[0], key, nvt[1]))
+            if script_args.show_ports:
+                print_without_pipe("PORT: %s" % (nvt[3]))
+            if script_args.descr:
+                print_without_pipe("DESCR: %s" % nvt[2])
+
+            if script_args.dfn and nvt[4]:
+                dfn_list = ", ".join(nvt[4])
+                if dfn_list:
+                    print_without_pipe("DFN-CERT: %s" % dfn_list)
+
+
+def end_session(im, msg, nagios_status):
+    """End the session
+
+    Close the socket if open and print the last msg
+
+    Arguments:
+        msg {string} -- Message to print
+        nagios_status {int} -- Exit status
+    """
+    print(msg)
+
+    # Delete this instance
+    im.delete_instance()
+
+    # Activate some waiting instances if possible
+    im.wake_instance()
+
+    # Close the connection to database
+    im.close_db()
+
+    sys.exit(nagios_status)
+
+
+def print_without_pipe(msg):
+    """Prints the message, but without any pipe symbol
+
+    If any pipe symbol is in the msg string, then it will be replaced with
+    broken pipe symbol.
+
+    Arguments:
+        msg {string} -- Message to print
+    """
+    if "|" in msg:
+        msg.replace("|", "¦")
+
+    print(msg)
+
+
+def pretty(xml):
```

Comment:
The whole function should be removed. It can be imported from python-gvm https://github.com/greenbone/python-gvm/blob/master/gvm/xml.py#L72  pretty_print function

`from gvm.xml import pretty_print as pretty` would just do it.